lib/rules/one-var.js

```diff
@@ -62,20 +120,46 @@ module.exports = function(context) {
         "FunctionDeclaration": startFunction,
         "FunctionExpression": startFunction,
         "ArrowFunctionExpression": startFunction,
+        "BlockStatement": startBlock,
```

Comment:
I've pushed up the code that enforces this, so the failing tests will be obvious.